src/core/segment_attributes.rs

```diff
@@ -0,0 +1,194 @@
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum SegmentAttribute {
+    ConjunctiveBool(bool),
+    DisjunctiveBool(bool),
+    StringList(Vec<String>),
+}
+
+impl SegmentAttribute {
+    pub fn merge(&mut self, other: &SegmentAttribute) {
+        match self {
+            SegmentAttribute::ConjunctiveBool(a) => match other {
+                SegmentAttribute::ConjunctiveBool(b) => *a = *a && *b,
+                _ => unreachable!(),
```

Comment:
surely unreachable is incorrect.

Reply:
Sure, I can reimplement in traits, we've discussed it above with @PSeitz 
Is the overall approach ok? I mean storing attributes in meta.json and introducing concept of attribute types containing implementation of their merge behavior?

Reply:
I've left implementation by enums and fixed unreachable